### PR TITLE
New Job list view

### PR DESCRIPTION
**Pull request by Gerhut**
Reimplment job list in fabric design

image
image
image

### CODE REVIEW

File: src/webportal/src/app/job/job-view/fabric/JobList/Ordering.js [hunk 1]
```diff
@@ -0,0 +1,64 @@
+import {getModified, getDuration, getStatusIndex} from './utils';
+
+export default class Ordering {
+  /**
+   * @param {"name" | "modified" | "user" | "duration" | “virtualCluster” | "status" | undefined} field
```
Comment (sunqinzheng) [inline src/webportal/src/app/job/job-view/fabric/JobList/Ordering.js]: The curved quotes may be typos

File: src/webportal/src/app/job/job-view/fabric/JobList/Ordering.js [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+      comparator = descending ? (a, b) => {
+        const da = getDuration(a);
+        const db = getDuration(b);
+        return da < db ? 1 : da > db ? -1 : 0;
+      } : (a, b) => {
+        const da = getDuration(a);
+        const db = getDuration(b);
+        return da > db ? 1 : da < db ? -1 : 0;
+      };
+    } else if (field === '“virtualCluster”') {
```
Comment (sunqinzheng) [inline src/webportal/src/app/job/job-view/fabric/JobList/Ordering.js]: The curved quotes may be typos

File: src/webportal/src/app/job/job-view/fabric/JobList/Ordering.js [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+
+  apply(jobs) {
+    const {field, descending} = this;
+    let comparator;
+    if (field == null) {
+      return jobs;
+    }
+    if (field === 'name') {
+      comparator = descending
+        ? (a, b) => (a.name < b.name ? 1 : a.name > b.name ? -1 : 0)
```
Comment (sunqinzheng) [inline src/webportal/src/app/job/job-view/fabric/JobList/Ordering.js]: You can use localeCompare to compare strings.
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/localeCompare

File: src/webportal/src/app/job/job-view/fabric/JobList/Table.jsx [hunk 4]
```diff
@@ -0,0 +1,185 @@
...
+        Succeeded: MessageBarType.success,
+        Failed: MessageBarType.error,
+        Stopped: MessageBarType.blocked,
+      }[statusText];
+      /** @type {import('@uifabric/styling').IStyle} */
+      const iconContainerStyle = {marginTop: 8, marginBottom: 8, marginLeft: 8};
+      /** @type {import('@uifabric/styling').IStyle} */
+      const textStyle = {marginTop: 8, marginRight: 8, marginBottom: 8};
+      return (
+        <div style={Object.assign(wrapperStyle, zeroPaddingRowFieldStyle)}>
```
Comment (sunqinzheng) [inline src/webportal/src/app/job/job-view/fabric/JobList/Table.jsx]: You can use column's classname to override cell's padding.
https://github.com/Microsoft/pai/blob/d13705706efe6659ab3b8fb6afc47232cc15198c/src/webportal/src/app/job/job-view/fabric/job-detail/components/task-role-container-list.jsx#L281

File: src/webportal/src/app/job/job-view/fabric/JobList/Table.jsx [hunk 5]
```diff
@@ -0,0 +1,182 @@
...
+          <MessageBar
+            messageBarType={messageBarType}
+            styles={{iconContainer: iconContainerStyle, text: textStyle}}
```
Comment (abuccts) [inline src/webportal/src/app/job/job-view/fabric/JobList/Table.jsx]: Show the same width for different status `MessageBar`s?

File: src/webportal/src/app/job/job-view/fabric/JobList/Ordering.js (hunk 1 above)
Comment (Gerhut) [reply src/webportal/src/app/job/job-view/fabric/JobList/Ordering.js]: It's just for better type intellisence.

https://www.typescriptlang.org/docs/handbook/type-checking-javascript-files.html#jsdoc-types-are-used-for-type-information
https://www.typescriptlang.org/docs/handbook/advanced-types.html#string-literal-types

**Pull request merged**